design-proposals/eim-decouple-microos-storage.md

```diff
@@ -0,0 +1,185 @@
+
+  dkam->>dkam: Generates and signs Micro-OS bundle
+
+  dkam->>pa: Store Micro-OS bundle on PV 
+
+  dkam->>dkam: Builds and signs iPXE EFI binary
+
+  dkam->>pa: Store signed iPXE EFI binary
+
+  user->>+rs: Download RS certificate & Micro-OS DER key
+  rs->>-user: [RS certificate, Micro-OS DER key]
+
+  user->>en: Load RS certificate & Micro-OS DER key to BIOS
+
+  note over en: EN powered on
+
+  en->>pa: Download iPXE binary
+  en->>en: EN boots into iPXE binary
+
+  en->>+rs: Download Micro-OS image
+  rs->>-en: [Micro-OS image]
+
+  en->>pa: Download the Micro-OS bundle
+
+  en->>en: Injects the Micro-OS bundle files to Micro-OS via initrd
+  
+  en->>en: Loads initramfs via initrd
+
+  en->>en: Boots into Micro-OS
+
+  note over user,rs: EN continues Onboarding and Provisioning
+```
+
+Steps 1-3: The CI/CD pipeline of Micro-EMT build the image and publishes it to the Release Service (S3 bucket), along with the signing key.
+The process is the same as currently used for the EMT image.
+
+Steps 4-8: At deployment time, DKAM reads the orchestrator CA certificate and cluster-wide variables,
+and generates two artifacts: 1) the signed Micro-OS bundle, and 2) the signed iPXE script that downloads 
+the Micro-OS image and the Micro-OS bundle.
+We assume that both artifacts are signed with the same key that is generated by DKAM.
+Note that iPXE includes standard public certificates (like Microsoft or AWS) so we don't need to embed
+these certificates, as long as it's not trusted by iPXE or a self-signed certificate that is used by a local storage.
+
+Steps 9-11: User needs to download a set of certificates and signing keys and upload them to BIOS. The set should include:
+- the orchestrator CA certificate (uploaded now)
+- the secure key used to sign iPXE and orchestrator bundle (uploaded now)
+- the secure key used to sign EMT (uploaded now) - only if EMT is used
+- the secure key used to sign Micro-OS image (new). This key is downloaded from the external storage.
+
+Steps 12-19: EN powers on and boots into the iPXE binary. The iPXE script downloads the Micro-OS from the
+URL that is provided by orchestrator and injected to the iPXE script by DKAM. Next, the iPXE script downloads
+the Micro-OS bundle from the Provisioning Nginx. Both initramfs of Micro-OS and Micro-OS bundle are loaded to
+the in-memory OS via initrd and the vmlinuz is used to boot into the Micro-OS. Once the Micro-OS is booted,
+
+## Rationale
+
+[A discussion of alternate approaches that have been considered and the trade
+offs, advantages, and disadvantages of the chosen approach.]
+
+## Affected components and Teams
+
+## Implementation plan
+
+**Targeted release: 3.1.**
+
+**Phase 1**: Work on a PoC by storing the current non-curated HookOS on the RS file server. The HookOS image is currently distributed as OCI artifact.
+The PoC would require to store it as non-OCI on the S3 bucket. The PoC should give an answer if the solution described in this proposal is feasible.
```

Comment:
can we keep it as OCI and use services as ECR ?

Reply:
Nope, we cannot download OCI scripts with iPXE